#__Bagging__
Now, let us see how bagging works.

## Step 1: Import Required Libraries

- Import pandas, sklearn.model_selection, sklearn.ensemble, and sklearn.tree libraries


In [ ]:
import pandas
from sklearn import model_selection
from sklearn.ensemble import BaggingClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import classification_report

## Step 2: Load the Dataset

- Load the Pima Indians Diabetes dataset using the given URL and assign the names
- Check the head of the data using the head() method


In [ ]:
url="https://raw.githubusercontent.com/jbrownlee/Datasets/master/pima-indians-diabetes.data.csv"

names = ['preg', 'plas', 'pres', 'skin', 'test', 'mass', 'pedi', 'age', 'class']
data = pandas.read_csv(url, names=names)
data.head()

preg  plas  pres  skin  test  mass   pedi  age  class
0     6   148    72    35     0  33.6  0.627   50      1
1     1    85    66    29     0  26.6  0.351   31      0
2     8   183    64     0     0  23.3  0.672   32      1
3     1    89    66    23    94  28.1  0.167   21      0
4     0   137    40    35   168  43.1  2.288   33      1

__Observations:__
- This is the head of the dataset.
- Create a DataFrame called data, which takes the data from the URL mentioned along with the names
- Create a feature matrix X and a target vector Y
- Assign all the rows up to the 8th column of the data to X
- Assign the 9th column of the same data to Y

In [ ]:
array = data.values
X = array[:,0:8]
Y = array[:,8]

## Step 3: Create a Decision Tree Classifier and a Bagging Classifier

- Set up the k-fold cross-validation with n_splits to 10
- Create a decision tree classifier named cart
- The estimator will be set to 100, which is equal to the number of trees.


In [ ]:
kfold = model_selection.KFold(n_splits=10)
cart = DecisionTreeClassifier()
num_trees = 100

Now, let's apply the bagging technique.
- Create a bagging classifier with the decision tree classifier (cart) as the base estimator
- The estimator will be set to 100, the number of trees.

In [ ]:
model = BaggingClassifier(estimator=cart,n_estimators=num_trees, random_state=12)

results = model_selection.cross_val_score(model, X, Y, cv=kfold)

print(results.mean())

0.7668831168831168


__Observations:__
- The mean cross-validation score is 0.76688.
- We have an accuracy of 76%.